fuzz/src/bin/gen_target.sh

```diff
@@ -20,6 +20,7 @@ GEN_TEST refund_deser
 GEN_TEST router
 GEN_TEST zbase32
 GEN_TEST indexedmap
+GEN_TEST onion_hop_data
```

Comment:
`msg_target`s get regenerated in CI using `msg_target_template.txt`. This results in a CI failure because there's a resulting `git diff` between the actual and generated fuzz tests, since the autogenerated test expects to use `test_msg_simple`, which doesn't work for the new onion hop data structs.

Reply:
Ah okay, we'd have to add a new macro because all 3 of those require that their structs implement `Writeable`. Would you prefer that?

Reply:
> Oh, I guess test_msg_simple is the least strict, I'm not sure I understand why we can't use that for the new onion structs? They should absolutely round-trip just fine.

The new `InboundOnionPayload` only implements `Readable`, and the new `OutboundOnionPayload` only implements `Writeable`.